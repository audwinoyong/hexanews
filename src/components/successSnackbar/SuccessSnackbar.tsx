/**
 * Display success snackbar row component.
 */

import * as React from 'react'
import Snackbar from '@material-ui/core/Snackbar'
import CheckCircleIcon from '@material-ui/icons/CheckCircle'
import SnackbarContent from '@material-ui/core/SnackbarContent'

import styles from './styles'

type Props = {
  isSnackbarOpen: boolean,
  message: string,
  handleClose: () => any,
}

const SuccessSnackbar: React.StatelessComponent<Props> = ({
  isSnackbarOpen,
  message,
  handleClose,
}) => {
  return (
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      open={isSnackbarOpen}
      onClose={handleClose}
      autoHideDuration={3000}
    >
      <SnackbarContent
        style={styles.successContainer}
        message={
          <span id='client-snackbar' style={styles.message}>
            <CheckCircleIcon style={styles.checkIcon} />
            {message}
          </span>
        }
      />
    </Snackbar>
  )
}

export default SuccessSnackbar