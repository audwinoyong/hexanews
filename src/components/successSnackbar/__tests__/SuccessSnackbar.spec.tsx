/**
 * Unit tests for checking the behavior of SuccessSnackbar components
 */

import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

import SuccessSnackbar from '../SuccessSnackbar'

Enzyme.configure({ adapter: new Adapter() })

describe('Success snackbar Row', () => {
  it('renders Success snackbar Row', () => {
    const component = renderer.create(
      <SuccessSnackbar
        isSnackbarOpen={false}
        message='a'
        handleClose={jest.fn()}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})