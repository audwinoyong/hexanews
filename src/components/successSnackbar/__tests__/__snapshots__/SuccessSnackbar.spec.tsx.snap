// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success snackbar Row renders Success snackbar Row 1`] = `null`;
